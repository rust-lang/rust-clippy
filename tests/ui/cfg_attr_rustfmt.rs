#![warn(clippy::deprecated_cfg_attr)]

// #![rusfmt::skip] requires https://github.com/rust-lang/rust/issues/54726
#![cfg_attr(rustfmt, rustfmt_skip)]

#[rustfmt::skip]
fn foo(
) {}

#[cfg_attr(rustfmt, rustfmt_skip)]
//~^ deprecated_cfg_attr
fn outer() {

    #[cfg_attr(rustfmt, rustfmt_skip)]
    //~^ deprecated_cfg_attr
    let _ = 1;

    #[cfg_attr(rustfmt, rustfmt_skip)]
    //~^ deprecated_cfg_attr
    foo();
}

fn invalid() {
    #![cfg_attr(rustfmt, rustfmt_skip)]

    #[cfg_attr(rustfmt, rustfmt::skip)]
    {}

    #[cfg_attr(rustfmt, rustfmt_skip)]
    foo()
}

#[clippy::msrv = "1.29"]
#[cfg_attr(rustfmt, rustfmt::skip)]
fn msrv_1_29() {}

#[clippy::msrv = "1.30"]
#[cfg_attr(rustfmt, rustfmt::skip)]
//~^ deprecated_cfg_attr
fn msrv_1_30() {}
