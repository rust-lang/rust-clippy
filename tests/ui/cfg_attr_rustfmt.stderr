error: `cfg_attr` is deprecated for rustfmt
  --> tests/ui/cfg_attr_rustfmt.rs:10:1
   |
LL | #[cfg_attr(rustfmt, rustfmt_skip)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::deprecated-cfg-attr` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::deprecated_cfg_attr)]`
help: use the `rustfmt::skip` tool attribute instead
   |
LL - #[cfg_attr(rustfmt, rustfmt_skip)]
LL + #[rustfmt::skip]
   |

error: `cfg_attr` is deprecated for rustfmt
  --> tests/ui/cfg_attr_rustfmt.rs:14:5
   |
LL |     #[cfg_attr(rustfmt, rustfmt_skip)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use the `rustfmt::skip` tool attribute instead
   |
LL -     #[cfg_attr(rustfmt, rustfmt_skip)]
LL +     #[rustfmt::skip]
   |

error: `cfg_attr` is deprecated for rustfmt
  --> tests/ui/cfg_attr_rustfmt.rs:18:5
   |
LL |     #[cfg_attr(rustfmt, rustfmt_skip)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use the `rustfmt::skip` tool attribute instead
   |
LL -     #[cfg_attr(rustfmt, rustfmt_skip)]
LL +     #[rustfmt::skip]
   |

error: `cfg_attr` is deprecated for rustfmt
  --> tests/ui/cfg_attr_rustfmt.rs:38:1
   |
LL | #[cfg_attr(rustfmt, rustfmt::skip)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: use the `rustfmt::skip` tool attribute instead
   |
LL - #[cfg_attr(rustfmt, rustfmt::skip)]
LL + #[rustfmt::skip]
   |

error: aborting due to 4 previous errors

